import "common.proto";

// 获取公会排行列表 command：810
message GetGuildRankRequest{
    required int32 rank_type = 1;//1获取排行列表，2获取推荐列表
    required int32 min_rank = 2;
    required int32 max_rank = 3;
}

// 获取公会排行列表 command：810
message GetGuildRankResponse{
    required CommonResponse res = 1;
    repeated GuildRank guild_rank = 2;
    optional int32 flag = 3;  //有没有军团 默认1 无军团1  满0
}

message GuildRank
{
    required int32 g_id = 1;    //id
    required int32 rank = 2;    //排名
    required bytes name = 3;    //名
    required int32 level = 4;   //等级
    required string president = 5;   //会长
    required int32 p_num = 6;   //人数
    required int32 icon_id = 7;   //军团徽章
    required bytes call = 8;   //公告
    optional int32 be_apply = 9;   //有没有被我申请
}

// 加入公会 command：802
message JoinGuildRequest{
	required int32 g_id = 1;  //公会id
}

message JoinGuildResponse{
    required CommonResponse res = 1;
    optional int32 spare_time = 4;
}

// 退出公会 command：803
message ExitGuildResponse{
    required CommonResponse res = 1;
}

// 搜索军团 command：
message SelectGuildRequest{
	required int32 select_type = 1;  //搜索类型
	required string value = 2;  //搜索值
}

message SelectGuildResponse{
    required CommonResponse res = 1;
    optional GuildRank guild_rank = 2;
}

// 创建公会 command：801
message CreateGuildRequest{
	required bytes name = 1;  //公会名
	required int32 icon_id = 2;  //军团图标
}

// 创建公会 command：801
message CreateGuildResponse{
    required CommonResponse res = 1;
}

// 获取公会信息 command：812
message GetGuildInfoResponse{
    required CommonResponse res = 1;

    optional int32 g_id = 2;    //id
    optional string name = 3;    //名
    optional int32 member_num = 4;   //人数
    optional int32 contribution = 6;     //军团的建设值
    optional int32 all_contribution = 5;     //军团的建设值
    optional string call = 7;    //公告
    optional int32 icon_id = 8;   //军团徽章
    optional int32 all_zan_num = 30;   // 军团今日已经赞的次数
    optional int32 zan_money = 29;   // 军团今日已经赞的次数

    optional string captain_name = 9;   //团长name
    optional int32 captain_level = 10;   //团长level
    optional int32 captain_power = 11;   //团长战力
    optional int32 captain_zan_num = 12;   //团长集赞次数
    optional int32 captain_vip_level = 16;   //团长vip等级
    optional int32 captain_icon = 17;   //团长头像
    optional int32 captain_zan_receive_state = 21;   //团长赞领取状态

    optional int32 zan_num = 13;   // 我今日已经赞的次数
    optional int32 last_zan_time = 27;   // 最后赞时间
    optional int32 position = 14;   //我的职位
    optional int32 have_apply = 15;   //有没有加军团申请 0 or 1

    optional int32 luck_num = 18;   //福运
    optional int32 bless_num = 19;   //祈福人数
    optional int32 bless_state = 20;   //今天有没有祈福
    repeated int32 bless_gift = 22;   //已经领取的祈福奖励

    optional int32 my_guild_rank = 23;
    optional int32 my_contribution = 24;  //我的剩余贡献值
    optional int32 my_all_contribution = 28;  //我的总贡献值
    optional int32 my_day_contribution = 25;  //我的今日共贡献值
    repeated BuildInfo build_info = 26;
}

message BuildInfo{
    required int32 build_type = 1;
    required int32 build_level = 2;
}

message UpBuildRequest{
    required int32 build_type = 1;
}

message UpBuildResponse{
    required CommonResponse res = 1;
}

// 修改公告 command：804
message EditorCallRequest{
	required bytes call = 1;  //公告内容
}

// 修改公告 command：804
message EditorCallResponse{
    required CommonResponse res = 1;
}


// 获取申请列表 command：813
message GetApplyListResponse{
    required CommonResponse res = 1;
    repeated ApplyUserInfo role_info = 2;
}

message ApplyUserInfo
{
    required int32 p_id = 1;    //id
    required string name = 2;    //名
    required int32 level = 3;   //等级
    required int32 vip_level = 4;   //vip等级
    required int32 fight_power = 5;   //战斗力
    required int32 is_online = 6;  //是否在线，1在线，0不在线
    optional int32 user_icon = 7;   //玩家头像
}

// 获取公会成员列表 command：811
message GetGuildMemberListResponse{
    required CommonResponse res = 1;
    repeated RoleInfo role_info = 2;
}

message RoleInfo
{
    required int32 p_id = 1;    //id
    required string name = 2;    //名
    required int32 level = 3;   //等级
    required int32 position = 4;   //职位
    required int32 day_contribution = 5;   //今日的贡献
    required int32 contribution = 6;   //剩余的贡献
    required int32 fight_power = 7;   //战斗力
    required int32 is_online = 8;  //是否在线，1在线，0不在线
    optional int32 user_icon = 9;   //玩家头像
    optional int32 vip_level = 16;   //vip等级
}

// 转让军团 command：806 change_president_806
message ChangePresidentRequest{
    required int32 p_id = 1;  //目标玩家id
}

// 转让军团 command：806 change_president_806
message ChangePresidentResponse{
    required CommonResponse res = 1;
}

// 处理加会申请 command：805
message DealApplyRequest{
	repeated int32 p_ids = 1;  //申请玩家id
	required int32 res_type = 2;  // 结果：1同意，2不同意,3清空
}
message DealApplyResponse{
    required CommonResponse res = 1;
    repeated int32 p_ids = 2;  //成功加入的玩家ID
}

// 踢出公会 command：807
message KickRequest{
	repeated int32 p_ids = 1;  //目标玩家ids
}

// 踢出公会 command：807
message KickResponse{
    required CommonResponse res = 1;
}

// 邀请加入军团command：1803
message InviteJoinRequest{
	required int32 user_id = 1;  //玩家id
}

// 邀请加入军团command：1803
message InviteJoinResponse{
    required CommonResponse res = 1;
}

// 处理 邀请加入军团command：1804
message DealInviteJoinRequest{
	required int32 res = 1;  //1 or 0 ，同意or 拒绝
    optional int32 guild_id = 2;  //guild id
    optional string mail_id = 3;
}

// 处理 邀请加入军团command：1804
message DealInviteJoinResResponse{
    required CommonResponse res = 1;
    optional int32 spare_time = 2;
}

// 赞一下 1807
message ZanResResponse{
    required CommonResponse res = 1;
    optional int32 zan_num = 2;   //今日赞数量
    optional GameResourcesResponse gain = 3;
}

// 团长领取  1806
message ReceiveResponse{
    required CommonResponse res = 1;
    optional GameResourcesResponse gain = 2;
}

// 祈福 command：809
message BlessRequest{
    required int32 bless_type = 1;
}

// 祈福 command：809
message BlessResponse{
    required CommonResponse res = 1;
}

// 领取祈福的奖励 command：1808
message GetBlessGiftRequest{
    required int32 gift_no = 1;
}

// 领取祈福的奖励 command：1808
message GetBlessGiftResponse{
    required CommonResponse res = 1;
    optional GameResourcesResponse gain = 2;
}

// 1809
message FindGuildRequest{
    optional bytes id_or_name = 1;
}

// 1809
message FindGuildResponse{
    required CommonResponse res = 1;
    optional GuildRank guild_info = 2;
}

// 1810
message AppointRequest{
    required int32 deal_type = 1; //1提拔2撤销
    required int32 p_id = 2;
}

// 1810
message AppointResponse{
    required CommonResponse res = 1;
}

// 1815
message PositionChange{
    required int32 position = 1;
}

// 获取圣兽模块信息
message GuildAnimalsInfoResponse{
    required int32 SkillPoints = 1;
    repeated GuildSkill guild_skill = 2;
    required int32 summon_times = 3;  // 军团已经召唤次数
    // required int32 attack_times = 4;  // 今日我打圣兽次数
    repeated GuildAnimal guild_animal = 5;
}

message GuildSkill{
    required int32 skill_type = 1;
    required int32 skill_level = 2;
}

message GuildAnimal{
    required int32 player_id = 1;
    required string player_name = 2;
    required int32 hp = 3;
    required int32 all_hp = 8;
    required string animal_id = 4;  // 唯一ID
    required int32 stage_id = 5;  // 关卡ID
    required int32 start_time = 6;
    required int32 last_time = 7;  // 我上次打的时间
}

// 召唤圣兽
message SummonGuildAnimalRequest{
    required int32 animal_type = 1;
}

// 召唤圣兽
message SummonGuildAnimalResponse{
    required CommonResponse res = 1;
    required GuildAnimal guild_animal = 2;
}

// 打圣兽
message AttackGuildAnimalRequest{
    required string animal_id = 4;  // 唯一ID
}

// 打圣兽
message AttackGuildAnimalResponse{
    required CommonResponse res = 1;
    required string animal_id = 2;  // 唯一ID
    required int32 hp = 3;
    optional GameResourcesResponse gain = 4;
}

// 军团技能升级
message UpGuildSkillRequest{
    required int32 skill_type = 1;
}

// 军团技能升级
message UpGuildSkillResponse{
    required CommonResponse res = 1;
}
